### PR TITLE
Remove pimple as a supported container

### PR DESCRIPTION
**Pull request by gsteel**
Remove support for Pimple due to the abandonment of `laminas-pimple-config`

### CODE REVIEW

**Comment by gsteel** [comment]
I'll have to come back to this later to resolve the psalm issues…

**Comment by froschdesign** [comment]
Related to https://github.com/mezzio/mezzio/issues/145

File: test/AppTest/Handler/HomePageHandlerFactoryTest.php [hunk 1]
```diff
@@ -56,10 +58,14 @@ public function testFactoryWithTemplate(): void
...
+                self::assertTrue(in_array($name, [
+                    RouterInterface::class,
+                    TemplateRendererInterface::class,
+                ]));
+
+                return true;
+            }))
             ->willReturnOnConsecutiveCalls(
                 $this->router,
                 $renderer
```
Comment (Xerkus) [inline test/AppTest/Handler/HomePageHandlerFactoryTest.php]: Imo, replace mock with a stub and use return map.

Comment (Xerkus) [reply test/AppTest/Handler/HomePageHandlerFactoryTest.php]: Or add return map since assumption about call older is brittle

**Review by Xerkus — APPROVED** [review]

**Pull request merged**